### ISSUE TITLE
Constructors compatible with Setfield.jl/Flatten.jl etc

### ISSUE THREAD

**Issue by rafaqz** [meggart/SentinelMissings.jl#3]
As usual I'm looking to reconstruct the `SentinalMissings` array object using ConstructionBase.jl, but it has type parameters not known from the fields.

I'm wondering if it would be possible to:

a. use a second field that holds a `Val{SV}()` object where `SV` is the sentinal value.
b. add a ConstructionBase.jl dep to add `constructorof` method that keeps the `SV` param.

Probably the first is preferable, as this package is so tiny.

**Comment by meggart** [comment]
Yes, I think option 1 would be no problem to implement. I can try next week.
